static/app/components/charts/optionCheckboxSelector.tsx

```diff
@@ -0,0 +1,186 @@
+import DropdownMenu from 'app/components/dropdownMenu';
+import Tooltip from 'app/components/tooltip';
+import Truncate from 'app/components/truncate';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import space from 'app/styles/space';
+import {SelectValue} from 'app/types';
+
+const defaultProps = {
+  menuWidth: 'auto',
+};
+
+type Props = {
+  options: SelectValue<string>[];
+  selected: string[];
+  onChange: (value: string[]) => void;
+  title: string;
+} & typeof defaultProps;
+
+type State = {
+  menuContainerWidth?: number;
+};
+
+class OptionCheckboxSelector extends Component<Props, State> {
+  static defaultProps = defaultProps;
+
+  state: State = {};
+
+  componentDidMount() {
+    this.setMenuContainerWidth();
+  }
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
+    return !isEqual(nextProps, this.props) || !isEqual(nextState, this.state);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.selected !== this.props.selected) {
+      this.setMenuContainerWidth();
+    }
+  }
+
+  setMenuContainerWidth() {
+    const menuContainerWidth = this.menuContainerRef?.current?.offsetWidth;
+    if (menuContainerWidth) {
+      this.setState({menuContainerWidth});
+    }
+  }
+
+  constructNewSelected(value: string) {
+    const {selected} = this.props;
+    // Cannot have no option selected
+    if (selected.length === 1 && selected[0] === value) {
+      return selected;
+    }
+    // Check if the value is already selected.
+    // Return a new updated array with the value either selected or deselected depending on previous selected state.
+    if (selected.includes(value)) {
+      return selected.filter(selectedValue => selectedValue !== value);
+    }
+    return [...selected, value];
+  }
+
+  menuContainerRef = createRef<HTMLDivElement>();
+
+  render() {
+    const {menuContainerWidth} = this.state;
+    const {options, onChange, selected, title, menuWidth} = this.props;
+    const selectedOption =
+      options.find(opt => selected.includes(opt.value)) || options[0];
+
+    return (
+      <InlineContainer>
+        <SectionHeading>{title}</SectionHeading>
+        <MenuContainer ref={this.menuContainerRef}>
+          <DropdownMenu alwaysRenderMenu={false} keepMenuOpen>
+            {({isOpen, getMenuProps, getActorProps}) => (
+              <Fragment>
+                <StyledDropdownButton {...getActorProps()} size="zero" isOpen={isOpen}>
+                  <TruncatedLabel>{String(selectedOption.label)}</TruncatedLabel>
+                </StyledDropdownButton>
+                <StyledDropdownBubble
+                  {...getMenuProps()}
+                  alignMenu="right"
+                  width={menuWidth}
+                  minWidth={menuContainerWidth}
+                  isOpen={isOpen}
+                  blendWithActor={false}
+                  blendCorner
+                >
+                  {options.map(opt => (
```

Comment:
should we add an index here?

Reply:
Hmm do you mean adding an index to be used as the key of the components from the list? Right now the key is `opt.value` which should always be defined and is should be unique since event view runs `uniqBy` on it. I think adding an index here shouldn't be necessary?

Reply:
ah yeah, I meant key and yup, missed the opt.value!